packages/core/src/sync/fragments.test.ts

```diff
@@ -1,107 +1,53 @@
 );
 
-test("buildLogFilterFragments generates 1 log filter fragment for null filter", () => {
-  const logFilterFragments = buildLogFilterFragments({
+test("getLogFilterFragmentIds generates 1 log filter fragment for null filter", () => {
+  const logFilterFragments = getLogFilterFragmentIds({
     type: "log",
     chainId: 1,
     address: undefined,
     topics: [null, null, null, null],
     includeTransactionReceipts: false,
   });
 
-  expect(logFilterFragments).toMatchObject([
-    {
-      id: "1_null_null_null_null_null_0",
-      address: null,
-      topic0: null,
-      topic1: null,
-      topic2: null,
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-  ]);
+  expect(logFilterFragments[0]!.id).toBe("log_1_null_null_null_null_null_0");
 });
 
-test("buildLogFilterFragments generates 1 log filter fragment for simple filter", () => {
-  const logFilterFragments = buildLogFilterFragments({
+test("getLogFilterFragmentIds generates 1 log filter fragment for simple filter", () => {
+  const logFilterFragments = getLogFilterFragmentIds({
     type: "log",
     chainId: 1,
     address: "0xa",
     topics: [null, null, null, null],
     includeTransactionReceipts: false,
   });
 
-  expect(logFilterFragments).toMatchObject([
-    {
-      id: "1_0xa_null_null_null_null_0",
-      address: "0xa",
-      topic0: null,
-      topic1: null,
-      topic2: null,
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-  ]);
+  expect(logFilterFragments[0]!.id).toBe("log_1_0xa_null_null_null_null_0");
 });
 
-test("buildLogFilterFragments generates 4 log filter fragment for 2x2 filter", () => {
-  const logFilterFragments = buildLogFilterFragments({
+test("getLogFilterFragmentIds generates 4 log filter fragment for 2x2 filter", () => {
+  const logFilterFragments = getLogFilterFragmentIds({
     type: "log",
     chainId: 115511,
     address: ["0xa", "0xb"],
     topics: [["0xc", "0xd"], null, "0xe", null],
     includeTransactionReceipts: false,
   });
 
-  expect(logFilterFragments).toMatchObject([
-    {
-      id: "115511_0xa_0xc_null_0xe_null_0",
-      address: "0xa",
-      topic0: "0xc",
-      topic1: null,
-      topic2: "0xe",
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-    {
-      id: "115511_0xa_0xd_null_0xe_null_0",
-      address: "0xa",
-      topic0: "0xd",
-      topic1: null,
-      topic2: "0xe",
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-    {
-      id: "115511_0xb_0xc_null_0xe_null_0",
-      address: "0xb",
-      topic0: "0xc",
-      topic1: null,
-      topic2: "0xe",
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-    {
-      id: "115511_0xb_0xd_null_0xe_null_0",
-      address: "0xb",
-      topic0: "0xd",
-      topic1: null,
-      topic2: "0xe",
-      topic3: null,
-      includeTransactionReceipts: 0,
-    },
-  ]);
+  expect(logFilterFragments[0]!.id).toBe("log_115511_0xa_0xc_null_0xe_null_0");
```

Comment:
Let's do one more review of these fragment ID encodings before we ship to main - want to make sure they are future-proofed for the more advanced filter types we've been chatting about.

Part of me thinks we should use a JSON design similar to the RPC request body encoding, and/or include a version key.